### PR TITLE
Start testing against AdoptOpenJDK

### PR DESCRIPTION
**Pull request by jasontedor**
This commit adds AdoptOpenJDK to the testing matrix.

### CODE REVIEW

File: .ci/java-versions.properties
Comment (jasontedor) [inline .ci/java-versions.properties]: This will be reverted before merging. The purpose here is to get a PR CI run before merging.

**Review by rjernst — APPROVED** [review]
LGTM

**Comment by rjernst** [comment]
Something is whacky with CI though:

```
Could not create task ':distribution:run#checkPrevious'.
> jps executable not found; ensure that you're running Gradle with the JDK rather than the JRE
```

Not only should jps be in adoptopenjdk, but I don't understand why `:distribution:run` would be trying to run; it's only meant for interactive usage and would cause the entire test run to block...

**Comment by jasontedor** [comment]
@rjernst Yeah, I am chasing that down internally. It looks like AdoptOpenJDK has not been integrated into the ephemeral workers yet, at least that's my current working theory.

**Comment by jasontedor** [comment]
@rjernst The reason that `:distribution:run#checkPrevious` is showing the failure here is because that task checks for `jps` in its configuration: https://github.com/elastic/elasticsearch/blob/567ad7f53ae40bbd7c6b4a8d59df93381bf8b8d4/buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy#L884-L915

**Comment by rjernst** [comment]
>  that task checks for jps in its configuration

Ouch. Ok, that will be fixed by the switch to testclusters. Completely unrelated to this PR.

**Comment by mark-vieira** [comment]
> The reason that :distribution:run#checkPrevious is showing the failure here is because that task checks for jps in its configuration

🤦‍♂ We can move that into a `gradle.taskGraph.whenReady { }` so this doesn't blow up for builds that don't run that task. Although it would still explode for any build that tried to bring up a cluster using `ClusterFormationTasks`.

**Comment by mark-vieira** [comment]
Not sure if it's something funny with our implementation but we can defer to Gradle to do this via `Jvm.current().getExecutable('jps')`. It's smart enough to handle if we are in the JRE directory and cross-platform bits.

**Comment by jasontedor** [comment]
```
Not sure if it's something funny with our implementation but we can defer to Gradle to do this via `Jvm.current().getExecutable('jps')`. It's smart enough to handle if we are in the JRE directory and cross-platform bits.
```

One thing to note here is that we are not getting `jps` from the currently executing Java (i.e., the JVM that Gradle is executing in) but for the runtime Java.

**Comment by jasontedor** [comment]
@mark-vieira For `jps` I opened #45677. In a follow-up I'll handle getting this out of configuration.

**Comment by jasontedor** [comment]
@elasticmachine update branch

**Comment by jasontedor** [comment]
@elasticmachine update branch

**Comment by jasontedor** [comment]
@elasticmachine update branch

**Pull request merged**
